### PR TITLE
gh-112278: Improve error handling in wmi module and tests

### PR DESCRIPTION
**Pull request by zooba**
* Issue: gh-112278

### CODE REVIEW

**Comment by zooba** [comment]
!buildbot .*windows.*

File: PC/_wmimodule.cpp [hunk 1]
```diff
@@ -279,9 +279,11 @@ _wmi_exec_query_impl(PyObject *module, PyObject *query)
     // a timeout.  The initEvent will be set after COM initialization, it will
     // take a longer time when first initialized.  The connectEvent will be set
     // after connected to WMI.
-    err = wait_event(data.initEvent, 1000);
     if (!err) {
-        err = wait_event(data.connectEvent, 100);
+        err = wait_event(data.initEvent, 1000);
+        if (!err) {
+            err = wait_event(data.connectEvent, 100);
+        }
```
Comment (aisk) [inline PC/_wmimodule.cpp:286]: We can reduce the nestinglike this:

```suggestion
        err = wait_event(data.initEvent, 1000);
    }
    if (!err) {
        err = wait_event(data.connectEvent, 100);
```

**Comment by aisk** [comment]
This change looks good to me except for the nitpick mentioned above. I tested it on my local machine and it passed.

But there were 3 changes to this file before, which haven't been backported to 3.12 yet, so I think Miss Islington will have trouble backporting this change. Should we just manually backport this change by pasting the whole file like you mentioned before in the original issue?

File: PC/_wmimodule.cpp (hunk 1 above)
Comment (zooba) [reply PC/_wmimodule.cpp:286]: Normally I'm a fan of reducing nesting, but I think in this case it better reflects that the two events are connected and if we spread it out it looks like they're unrelated.

**Comment by zooba** [comment]
Yeah, I expect this to require a manual backport. I'll update the entire file when I do it.

**Pull request merged**